src/test/java/dk/sdu/se_f22/sortingmodule/range/rangepublic/TimeFilterTest.java

```diff
@@ -0,0 +1,387 @@
+
+class TimeFilterTest {
+
+    /**
+     *The returned filter has: <br>
+     * min: 2018-11-30T15:35:24.00Z<br>
+     * max: 2022-11-30T15:35:24.00Z
+     *
+     * @param productAttribute can be either "publishedDate" or "expirationDate"
+     */
+    static TimeFilter getTestFilter(String productAttribute) {
+        Instant dbMin = Instant.parse("2018-11-30T15:35:24.00Z");
+        Instant dbMax = Instant.parse("2022-11-30T15:35:24.00Z");
+        return new TimeFilter(0, "test name", "test description", productAttribute, dbMin, dbMax);
+    }
+
+    @ParameterizedTest(name = "{0}")
+    @ValueSource(strings = {"publishedDate", "expirationDate"})
+    void getType(String productAttribute) {
+        TimeFilter timeFilter = getTestFilter(productAttribute);
+        assertEquals(FilterTypes.TIME, timeFilter.getType());
+    }
+
+    @Nested
+    @DisplayName("Use filter method")
+    class useFilterMethod {
+        // Make a test that ensures that changing productAttribute actually changes the attribute being filtered
+
+        @ParameterizedTest
+        @DisplayName("Changing product attribute actually changes attribute used for filtering")
+        @ValueSource(strings = {"publishedDate", "expirationDate"})
+        void changingProductAttributeActuallyChangesBeingFiltered(String productAttribute) throws IllegalMinMaxException {
+            // Creating filter with productattribute publishDate or experationDate
+            TimeFilter filter = getTestFilter(productAttribute);
+            Instant userMin = Instant.parse("2019-11-30T15:35:24.00Z");
+            Instant userMax = Instant.parse("2021-11-30T15:35:24.00Z");
+
+            filter.setUserMin(userMin);
+            filter.setUserMax(userMax);
+
+            // Preparing input list
+            List<Product> mockResults = Helpers.readMockProductResultsFromFile("rangepublic/ProductsForTimeFilterTest.csv", true);
+
+            // Preparing the expected result list
+            // Removes index one in the first if statement,
+            // because the productattribute is not inside the usermin and usermax range
+            List<Product> copy = List.copyOf(mockResults);
+            ArrayList<Product> expectedResults = new ArrayList<>(copy);
+            if(productAttribute.equals("publishedDate")) {
+                expectedResults.remove(1);
+            }
+            if(productAttribute.equals("expirationDate")) {
+                expectedResults.remove(0);
+            }
+
+            Collection<Product> filteredResults = filter.useFilter(mockResults);
+
+            Assertions.assertEquals(expectedResults,filteredResults);
+        }
+
+
+        @DisplayName("filter a list of actual products")
+        @ParameterizedTest(name = "{0}")
+        @MethodSource("useFilterArguments")
+    void useFilter(TimeFilter internalFilter) {
+            // preparing the input list
+            // Note: Helpers.readMockProductResultsFromFile Should be tested in a separate test.
```

Comment:
Note: Helpers.readMockProductResultsFromFile is expected to be tested in a separate test.
sounds better

Reply:
I expect this means Klaes has implemented the comment